2-collectors/scala-stream-collector/kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -148,11 +186,17 @@ class KinesisSink private (
   val RecordThreshold = bufferConfig.recordLimit
   val TimeThreshold = bufferConfig.timeLimit
 
-  private val maxBackoff = kinesisConfig.backoffPolicy.maxBackoff
-  private val minBackoff = kinesisConfig.backoffPolicy.minBackoff
-  private val randomGenerator = new java.util.Random()
-
   log.info("Creating thread pool of size " + kinesisConfig.threadPoolSize)
+  maybeSqs match {
+    case Some(sqs) =>
+      log.info(
+        s"SQS buffer for failed Kinesis stream '$streamName' is set up as: ${sqs.sqsBufferName}"
+      )
+    case None =>
+      log.warn(
+        s"No SQS buffer set up, all failed events from Kinesis stream '$streamName' will be dropped. (consider setting a SQS Buffer in config.hocon)"
```

Comment:
For surge protections it's useless. For Kinesis outage it might help, but only for a short while, because for real traffic this will use all the collectors memory quickly (due to how retries are implemented at the moment at least). When collector dies all the events it had been buffering are lost (it tries to send it, but I assume the outage doesn't stop exactly at the moment when collector dies and invokes the shutdown hook). We had discussion on slack (mostly with Josh and João, but on open channels) to use other mechanisms, like saving to disk, but finally the approach with sqs was the winner.